src/app/errorClasses.ts

```diff
@@ -0,0 +1,10 @@
+export class DatabaseError extends Error {
+    constructor(errorCategory: DatabaseOperation, faultyArea = "") {
```

Comment:
surprised faultyArea didn't want to be typed as a string

Reply:
Same tbf